File: app/src/main/java/hr/ml/moviesdiscover/util/UiListToString.java
package hr.ml.moviesdiscover.util;

import java.util.List;

import hr.ml.moviesdiscover.rest.model.GenreFromRetrofit;

public class UiListToString {

    public static String genresToString(List<GenreFromRetrofit> genres) {
        StringBuilder builder = new StringBuilder();
        builder.append(" | ");
        for (GenreFromRetrofit genre : genres) {
            builder.append(genre.getName());
            builder.append(", ");
        }
        String genresStr = builder.toString();

        return genresStr.substring(0, genresStr.length()-2);
    }

}

File: app/src/main/kotlin/hr/ml/moviesdiscover/util/UiListToString.kt
package hr.ml.moviesdiscover.util

import hr.ml.moviesdiscover.rest.model.GenreFromRetrofit

class UiListToString {

    companion object {
        private const val separator = " | "

        fun genresToString(genres: List<GenreFromRetrofit>): String {
            val builder = StringBuilder()

            builder.append(separator)

            for (genre in genres) {
                builder.append(genre.name + ", ")
            }

            val genresOutput = builder.toString()

            // trim ", " at the end
            return genresOutput.substring(0, genresOutput.length - 2)
        }

    }

}